### ISSUE TITLE
Dynamic add route

### ISSUE THREAD

**Issue by julijan** [sveltejs/kit#8078]
### Describe the problem

It would be nice having an alternative way to add routes. Not keen on file system routing, at least not in the fashion it's currently done in, main reason is that each route needs a +page.svelte, so if I have multiple pages open in my IDE, it's kind of confusing and hard to tell which page is which, in some cases I had 3 of those, with some +page.ts and +page.server.ts's, and it becomes quite easy to get lost.

### Describe the proposed solution

We could have /src/dynamicRoutes/[meaningfulFileName].ts that exports a function getting app as an argument, and dynamically add routes from there. We could for example create /src/dynamicRoutes/Auth.ts and add all auth related routes from that one file, and have one file for each "set" of routes. Then instead of having 3 non-unique and non-descriptive files open in the IDE, we could be looking at "Auth.ts" which is far more descriptive. I propose Svelte should load all files from /src/dynamicRoutes automatically once the server is started.

app.addRequestHandler(Array<RequestMethod>, matchPattern: string, { page: import('../somePage.svelte'), actions?: {...}, load?: Load|ServerLoad })

Example:
/src/dynamicRoutes/Auth.ts
`export default function(app: App) {

app.addRequestHandler(['GET'], matchPattern: '/auth/register', {
page: import('../components/RegisterForm.svelte'),
actions?: {
default: async function() {... handle registration form submission...}
},
load: async function() {...do something at load...}
})

app.addRequestHandler(['GET'], matchPattern: '/auth/login', {
page: import('../components/LoginForm.svelte'),
actions?: {
default: async function() {... handle login form submission...}
},
load: async function() {...do something at load...}
})

}

This doesn't need to replace the file system routing, but it could work in addition to it, allowing us to use either or the combination of those.

### Alternatives considered

An alternative solution to this problem would be making the Svelte VSC extension somehow highlight/colorize the open file tabs, but I feel it would be better to solve it at it's core.

### Importance

would make my life easier

**Comment by mustofa-id** [comment]
Doesn't the tab in VSC also show the directory name?

About routing, here's a [party](https://github.com/sveltejs/kit/discussions/5748) you might have missed.

**Comment by julijan** [comment]
It shows the directory name once you open multiple files with the same name. Also, the directory name is in small font and kind of hard to see, and +page.svelte is just taking up the space. I don't see a good reason for not allowing us to bind handlers for routes in a way similar to what I proposed, there are definitely arguments why it makes sense, but I can't think of a solid one against. One could be, what happens if someone binds multiple handlers to same request method/URL, or if someone binds a handler while at the same time that route is defined using the file system method, but both of those could be solved by just using the last one and giving preference to file system routes, while providing a warning that "there are multiple request handlers for [RequestMethod] [URL], only one will be used".

I started using Svelte the other day, so I did miss that discussion.
